app/static/src/app/components/sessions/SessionsPage.vue

```diff
@@ -101,6 +124,50 @@ export default defineComponent({
             toggleEditSessionLabelOpen(true);
         };
 
+        const ensureFriendlyId = async (session: SessionMetadata) => {
+            lastCopySessionId.value = session.id;
+            if (session.friendlyId) {
+                return session.friendlyId;
+            }
+            lastCopyMsg.value = "Fetching code...";
+            await store.dispatch(`sessions/${SessionsAction.GenerateFriendlyId}`, session.id);
+            await nextTick();
+            const friendlyId = sessionsMetadata.value.find((m: SessionMetadata) => m.id === session.id)?.friendlyId;
+            if (!friendlyId) {
+                lastCopyMsg.value = "Error fetching code";
+            }
+            return friendlyId;
+        };
+
+        const copyText = (text: string) => {
+            window.navigator.clipboard.writeText(text);
+            lastCopyMsg.value = `Copied: ${text}`;
+        };
+
+        const copyLink = async (session: SessionMetadata) => {
+            const friendlyId = await ensureFriendlyId(session);
+            if (friendlyId) {
+                const link = `${baseUrl.value}/apps/${appName.value}/?share=${friendlyId}`;
+                copyText(link);
+            }
```

Comment:
You are already checking if not friendly in the `ensureFriedlyId()` method you are calling above, I am curious why you are checking if friendly exists again ?

Reply:
This will only happen if there's been an error generating the friendly id (which we only attempt if it doesn't already exist, hence the earlier check). So we do need both.